core/src/main/java/io/confluent/rest/auth/AuthUtil.java

```diff
@@ -141,16 +141,27 @@ private static ConstraintMapping createConstraint(
       final boolean authenticate,
       final String pathSpec
   ) {
-    final Constraint constraint = new Constraint();
-    constraint.setAuthenticate(authenticate);
+
+    final Constraint.Builder constraint = new Constraint.Builder();
+
     if (authenticate) {
       final List<String> roles = restConfig.getList(RestConfig.AUTHENTICATION_ROLES_CONFIG);
-      constraint.setRoles(roles.toArray(new String[0]));
+      if (roles.isEmpty()) {
+        //equivalent of setting an empty <auth-constraint> if no setRoles(String []) is set,
+        // forbidding access
+        constraint.authorization(Constraint.FORBIDDEN.getAuthorization());
+      } else if (roles.size() == 1 && roles.get(0).equals("*")) {
```

Comment:
```suggestion
      } else if (roles.size() == 1) {
        if (roles.get(0).equals("*")) {
          constraint.authorization(Constraint.KNOWN_ROLE.getAuthorization());
        } else if (roles.get(0).equals("**")) {
          constraint.authorization(Constraint.ANY_USER.getAuthorization());
        }
```

We are using "**" to bypass Jetty Authorization (as we depend on MDS for authorization.
@apurushotama can you please make the suggested changes. Thanks!

Reply:
LGTM! from CP Security point of view. Thanks!